src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -76,29 +81,44 @@ public List<IexLastTradedPrice> getLastTradedPrice(
   @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
       MediaType.APPLICATION_JSON_VALUE})
   public IexHistoricalPrice getHistoricalPrice(
-      @RequestParam(value = "symbol") @NotNull final String symbols,
+      @RequestParam(value = "symbol") @NotNull @NotEmpty final String symbols,
       @RequestParam(value = "range", required = false) final Optional<@Pattern.List(
           {@Pattern(regexp = ".*"), @Pattern(regexp = VALID_DATES)}) String> range,
       @RequestParam(value = "date", required = false) final Optional<Integer> date) {
     IexHistoricalPrice price;
-//@Pattern.List({@Pattern(regexp =".*"), @Pattern (regexp = VALID_DATES)})
+
     //3 Parameters: Symbol is REQUIRED, Date/Range are optional and cannot both be present
 
     if (date.isPresent()) {
+      //yyyyMMdd
       int dateVal = date.get();
 
       //Max validation depends on current time:
-      if (dateVal >= System.currentTimeMillis()) {
-        throw new IllegalArgumentException("Cannot get a price for a time in the future");
+      long timeOfNow = Long.parseLong(LocalDate.now().toString().replace("-", ""));
+      if (timeOfNow < dateVal) {
+        throw new IllegalArgumentException("Cannot retrieve a price from the future");
       }
 
       //Min validation depends on current time:
       long fiveYearsAgo = Long.parseLong(LocalDate.now().toString().replace("-", ""));
-      fiveYearsAgo -= 500000;
-      if (dateVal <= fiveYearsAgo) {
-        throw new IllegalArgumentException(
-            "Cannot go back more than 5 years: " + "Expected > " + fiveYearsAgo + " was: "
-                + dateVal);
+
+      fiveYearsAgo -= 50000;
+
+      //In test: Dummy value. In Dev: fiveYearsAgo
+      //Unix Timestamp to LocalDate: https://stackoverflow.com/questions/35183146/how-can-i-create-a-java-8-localdate-from-a-long-epoch-time-in-milliseconds
+      LocalDate fiveYearsAgoDate =
+          Instant.ofEpochMilli(fiveYearsAgo).atZone(ZoneId.systemDefault()).toLocalDate();
+
+      DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd");
```

Comment:
"We can initialize this as a class-level constant" referred to the `DateTimeFormatter formatter`. The fiveYearsAgoDate should still be computed each time, for the reasons you described.
Declaring the formatter as a class-level field helps a bit with performance and memory, since there is a cost to creating a new instance each time this method is called. Probably not a massive difference in a practical sense, but at the same time it's a super easy fix. Also makes it easy to reuse if we add more endpoints in the future which also need date parsing.